hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ZKReplicationQueueStorage.java

```diff
@@ -417,6 +417,11 @@ public long getWALPosition(ServerName serverName, String queueId, String fileNam
       }
       // add delete op for peer
       listOfOps.add(ZKUtilOp.deleteNodeFailSilent(oldQueueNode));
+      // Append new queue id for prevent lock competition in zookeeper server.
+      String claimLockZNode = ZNodePaths.joinZNode(queuesZNode, "cversion_" + newQueueId);
+      // update cversion for queuesZNode
+      listOfOps.add(ZKUtilOp.createAndFailSilent(claimLockZNode, HConstants.EMPTY_BYTE_ARRAY));
```

Comment:
So this is the trick. Better speak more clearly that this is for changing the cversion of the root queueZNode, and add a link to the getAllWALs method.

Reply:
You are right, I added more comments and documentation to explain why we update the root queuesZNode and the relationship between claimQueue and getAllWALs